### PR TITLE
Support Arrays and ignore if statements

### PR DESCRIPTION
**Pull request by timtebeek**
Pulled out of #53 for easier review and to minimize diff.

### CODE REVIEW

File: src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java [hunk 1]
```diff
@@ -811,17 +806,16 @@ private boolean resolve(Context context, JCCompilationUnit cu) {
...
-    private final Set<String> printedMessages = new HashSet<>();
+    private final Map<String, Integer> printedMessages = new TreeMap<>();
 
     /**
      * @param message The message to print
      * @param symbol  The symbol to attach the message to; printed as clickable link to file
      */
     private void printNoteOnce(String message, Symbol.ClassSymbol symbol) {
-        if (printedMessages.add(message)) {
+        if (printedMessages.compute(message, (k, v) -> v == null ? 1 : v + 1) == 1) {
```
Comment (pstreef) [inline src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java:818]: why this change? Is this somehow more performant?

**Review by pstreef — APPROVED** [review]

File: src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java (hunk 1 above)
Comment (timtebeek) [reply src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java:818]: It's to allow one to inspect the number of times recipes are excluded, as used in this https://github.com/openrewrite/rewrite-templating/issues/47#issuecomment-1868511830

Didn't want to make that part of the standard output though; I think once and linked to the source is enough, as opposed to just a summary of how often there are exclusions not tied to a particular instance. Having this counter allows us to easily print it as a one off though.
